atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1341,80 +1343,56 @@ private void commitWrites(TransactionService transactionService) {
-                                SafeArg.of("microsForPunch", microsForPunch),
-                                SafeArg.of("microsForReadWriteConflictCheck", microsForReadWriteConflictCheck),
-                                SafeArg.of("microsForPutCommitTs", microsForPutCommitTs),
-                                SafeArg.of("microsForPreCommitLockCheck", microsForPreCommitLockCheck),
-                                SafeArg.of("microsForUserPreCommitCondition", microsForUserPreCommitCondition),
-                                SafeArg.of("microsForCommitStage", microsForCommitStage),
-                                SafeArg.of("microsSinceCreation", microsSinceCreation),
-                                tableRefs.safeTableRefs(),
-                                tableRefs.unsafeTableRefs())
-                        .build();
-                    });
+            optionalProfile = Optional.of(ImmutableTransactionCommitProfile.builder()
+                    .startTimestamp(getTimestamp())
+                    .acquireRowLocksMicros(microsForRowLocks)
+                    .conflictCheckMicros(microsCheckingForConflicts)
+                    .writingToSweepQueueMicros(microsWritingToTargetedSweepQueue)
+                    .keyValueServiceWriteMicros(microsForWrites)
+                    .commitTimestampMicros(microsForGetCommitTs)
+                    .punchMicros(microsForPunch)
+                    .readWriteConflictCheckMicros(microsForReadWriteConflictCheck)
+                    .verifyPreCommitLockCheckMicros(microsForPreCommitLockCheck)
+                    .verifyUserPreCommitConditionMicros(microsForUserPreCommitCondition)
+                    .putCommitTimestampMicros(microsForPutCommitTs)
+                    .commitTimestamp(commitTimestamp)
+                    .build());
         } finally {
-            runAndGetDurationMicros(() -> timelockService.unlock(ImmutableSet.of(commitLocksToken)),
-                    "postCommitUnlock");
-        }
-    }
-
-    private void updateNonPutOverheadMetrics(long microsWritingToTargetedSweepQueue, long microsForWrites,
-            long microsForCommitStage) {
-        long nonPutOverhead = microsForCommitStage - microsForWrites - microsWritingToTargetedSweepQueue;
-        getTimer("nonPutOverhead").update(nonPutOverhead, TimeUnit.MICROSECONDS);
+            long microsForPostCommitUnlock = runAndReportTimeAndGetDurationMicros(
+                    () -> timelockService.unlock(ImmutableSet.of(commitLocksToken)), "postCommitUnlock");
 
-        // Dropwizard Metrics doesn't support histograms of double yet, so using longs as a workaround
-        getHistogram("nonPutOverheadMillionths").update(
-                Math.round((1_000_000. * nonPutOverhead) / microsForCommitStage));
+            // We only care about detailed profiling for successful transactions
+            optionalProfile.ifPresent(profile -> profileProcessor.consumeProfilingData(
```

Comment:
This part appeared slightly sketchy (you get the profile and a few random extra bits), but it's ok, especially as the name (TransactionCommitProfile) is accurate and descriptive.